pkg/controller/config/config_controller.go

```diff
@@ -256,10 +265,23 @@ func (r *ReconcileConfig) replayData(ctx context.Context, w *watch.Set) error {
 		}
 
 		for i := range u.Items {
+			syncKey := strings.Join([]string{u.Items[i].GetNamespace(), u.Items[i].GetName()}, "/")
+
 			if _, err := r.opa.AddData(context.Background(), &u.Items[i]); err != nil {
+				r.syncMetricsCache.AddObject(syncKey, syncc.Tags{
+					Kind:   u.Items[i].GetKind(),
+					Status: metrics.ErrorStatus,
+				})
 				return fmt.Errorf("adding data for %+v: %w", gvk, err)
 			}
+
+			r.syncMetricsCache.AddObject(syncKey, syncc.Tags{
+				Kind:   u.Items[i].GetKind(),
+				Status: metrics.ActiveStatus,
+			})
 		}
+
+		r.syncMetricsCache.ReportSync(&syncc.Reporter{Ctx: context.TODO()})
```

Comment:
Do we want to run this as under a `defer` statement to avoid the case where we misreport metrics because we return early due to an error calling `r.opa.AddData()`?

Reply:
missed this, thanks. added